api/filters/catalog_filter_by_visibility.go

```diff
@@ -0,0 +1,183 @@
+package filters
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/Peripli/service-manager/api/osb"
+
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+)
+
+const CatalogFilterByVisibilityName = "CatalogFilterByVisibility"
+
+func NewCatalogFilterByVisibility(repository storage.Repository) *CatalogFilterByVisibility {
+	return &CatalogFilterByVisibility{
+		repository: repository,
+	}
+}
+
+type CatalogFilterByVisibility struct {
+	repository storage.Repository
+}
+
+func (vf *CatalogFilterByVisibility) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	ctx := req.Context()
+	userCtx, ok := web.UserFromContext(ctx)
+	if !ok {
+		return nil, errors.New("no user found")
+	}
+	if userCtx.AuthenticationType != web.Basic {
+		log.C(ctx).Debugf("Authentication is %s, not basic so proceed without visibility filter criteria", userCtx.AuthenticationType)
+		return next.Handle(req)
+	}
+	platform := &types.Platform{}
+	if err := userCtx.Data(platform); err != nil {
+		return nil, err
+	}
+	if platform.Type != k8sPlatformType {
+		log.C(ctx).Debugf("Platform type is %s, which is not kubernetes. Skip filtering on visibilities", platform.Type)
```

Comment:
this is a an if that is probably going to be refactored pretty soon due to changing scenarios - i just want to make sure that we have concrete test cases that verify that after refactoring the behaviour is the same (in this case it would be (when fetching catalog for cloudfoundry platform and there are no visibilities for service x plan y, it is still part of the returned catalog)